### PR TITLE
Respect Windows job object's CPU quota

### LINKED ISSUE #46889: Container limits not supported by .NET runtime in process isolation mode -- Windows containers
We intend to support container limits on both Linux and Windows. It appears that we don't have support for process-isolated containers on Windows. That should be resolved.

In particular, both CPUs and memory are not being honored for process-isolated Windows containers. They should behave the same way as Linux containers. We need to fix that. Let's start with .NET 6.0, and then look at backporting needs.

Command used in each example:

```console
docker run --rm -p 8080:80 -m 80 --cpus 1 mcr.microsoft.com/dotnet/samples:aspnetapp
```

Windows process-isolated behavior:

image

Note: My local machine is on insider builds, and so doesn't support process-isolated containers of any kind. This example is from a Windows Server 2019 Azure VM. The examples below are from my local machine. That's why they look different.

Windows Hyper-V-isolated behavior:

image

Linux behavior:

image

### PR DESCRIPTION
**Pull request by AntonLapounov**
Fixes #46889.

### CODE REVIEW

File: src/coreclr/utilcode/util.cpp [hunk 1]
```diff
@@ -1321,6 +1321,36 @@ int GetCurrentProcessCpuCount()
...
+            {
+                maxRate = cpuRateControl.MaxRate;
+            }
+
+            // The rate is the percentage times 100
+            const DWORD MAXIMUM_CPU_RATE = 10000;
+
+            if (0 < maxRate && maxRate < MAXIMUM_CPU_RATE)
+            {
+                DWORD cpuLimit = (maxRate * GetTotalProcessorCount() + MAXIMUM_CPU_RATE - 1) / MAXIMUM_CPU_RATE;
```
Comment (mangod9) [inline src/coreclr/utilcode/util.cpp:1348]: assume have you validated that this formula gets the right value?

File: src/coreclr/utilcode/util.cpp [hunk 2]
```diff
@@ -1321,6 +1321,36 @@ int GetCurrentProcessCpuCount()
                     count = 64;
             }
         }
+
+        JOBOBJECT_CPU_RATE_CONTROL_INFORMATION cpuRateControl;
```
Comment (mangod9) [inline src/coreclr/utilcode/util.cpp:1325]: Perhaps add some comments here for clarity as to why the JOB object is being queried?

**Comment by mangod9** [comment]
@richlander as FYI.

File: src/coreclr/utilcode/util.cpp (hunk 2 above)
Comment (AntonLapounov) [reply src/coreclr/utilcode/util.cpp:1325]: There is a comment at the top of this function: "take into account process affinity and CPU quota limit". For Windows we used to ignore the CPU quota limit, which is now enforced right after handling the process affinity, thus fulfilling the existing comment.

File: src/coreclr/utilcode/util.cpp (hunk 1 above)
Comment (AntonLapounov) [reply src/coreclr/utilcode/util.cpp:1348]: Yes.

**Review by mangod9 — APPROVED** [review]

**Pull request merged**